src/Notepads/Controls/TextEditor/TextEditor.xaml.cs

```diff
@@ -783,17 +783,31 @@ public void CutSelectedTextToWindowsClipboard(TextControlCuttingToClipboardEvent
                 args.Handled = true;
             }
 
-            CopySelectedTextToWindowsClipboardInternal();
+            CopyTextToWindowsClipboardInternal(false);
             TextEditorCore.Document.Selection.SetText(TextSetOptions.None, string.Empty);
         }
 
-        private void CopySelectedTextToWindowsClipboardInternal()
+        private void CopyTextToWindowsClipboardInternal(bool clearLineSelection)
         {
             try
             {
                 DataPackage dataPackage = new DataPackage { RequestedOperation = DataPackageOperation.Copy };
+                var isTextSelected = TextEditorCore.Document.Selection.Length > 0;
+                var cursorPosition = TextEditorCore.Document.Selection.StartPosition;
+
+                if (!isTextSelected)
+                {
+                    TextEditorCore.Document.Selection.Expand(TextRangeUnit.Paragraph);
+                }
+
                 var text = LineEndingUtility.ApplyLineEnding(TextEditorCore.Document.Selection.Text, GetLineEnding());
                 dataPackage.SetText(text);
+
+                if (clearLineSelection && !isTextSelected)
```

Comment:
move this logic to CutSelectedTextToWindowsClipboard and remove the "bool clearLineSelection" parameter.

Reply:
okay, but if I move this logic to CutSelectedTextToWindowsClipboar then there will be code duplication, because "Copy" also needs it.

Reply:
CopyTextToWindowsClipboardInternal will do the copy, isn't it?